### PR TITLE
Google Pay button

### PR DESCRIPTION
**Pull request by OscarSpruit**
## Description
Add option to display the Google Pay button inside of the Google Pay component. For drop-in it will not display.

## Checklist 
- [x] PR is labelled 
- [x] Code is unit tested
- [x] Changes are tested manually

COAND-942

### CODE REVIEW

File: googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/model/GooglePayComponentParamsMapper.kt [hunk 1]
```diff
@@ -79,6 +80,17 @@ internal class GooglePayComponentParamsMapper(
...
     }
 
+    private fun shouldShowSubmitButton(
+        commonComponentParams: CommonComponentParams,
+        googlePayConfiguration: GooglePayConfiguration?,
+    ): Boolean {
+        return if (commonComponentParams.isCreatedByDropIn) {
+            false
+        } else {
+            googlePayConfiguration?.isSubmitButtonVisible ?: false
```
Comment (jreij) [inline googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/model/GooglePayComponentParamsMapper.kt:90]: Can we add a todo to change this to `true` by default in v6?

File: googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/DefaultGooglePayDelegate.kt [hunk 2]
```diff
@@ -154,7 +142,7 @@ internal class DefaultGooglePayDelegate(
         )
     }
 
-    @Deprecated("Deprecated in favor of startGooglePayScreen()", replaceWith = ReplaceWith("startGooglePayScreen()"))
+    @Deprecated("Deprecated in favor of onSubmit()", replaceWith = ReplaceWith("onSubmit()"))
```
Comment (jreij) [inline googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/DefaultGooglePayDelegate.kt:145]: Do we need to deprecate internal functions?

File: googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/GooglePayButtonView.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+) : PayButton(context, attrs, defStyleAttr) {
+
+    private val binding = ViewGooglePayButtonBinding.inflate(LayoutInflater.from(context), this)
+
+    override fun initialize(delegate: ButtonDelegate) {
+        check(delegate is GooglePayDelegate)
+
+        binding.payButton.initialize(
+            ButtonOptions.newBuilder()
+                .setButtonType(ButtonType.PAY)
```
Comment (jreij) [inline googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/GooglePayButtonView.kt:33]: Will you add the button UI styling configuration in another PR?

Comment (OscarSpruit) [reply googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/GooglePayButtonView.kt:33]: Yes

File: googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/DefaultGooglePayDelegate.kt (hunk 2 above)
Comment (OscarSpruit) [reply googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/DefaultGooglePayDelegate.kt:145]: Not necessary, but then we know we can clean it up later

**Review by jreij — APPROVED** [review]

**Pull request merged**